cds_migrator_kit/rdm/migration/transform/transform.py

```diff
@@ -304,21 +311,54 @@ def creator_identifiers(creator):
             inner_dict = creator.get("person_or_org", {})
             identifiers = inner_dict.get("identifiers", [])
             for identifier in identifiers:
-                # TODO process CDS and CERN Ids when names vocabulary ready
-                if identifier["scheme"] == "inspire":
+                # we check for unknown schemes
+                if identifier["scheme"] in ["inspire", "orcid", "cern", "cds"] :
                     processed_identifiers.append(identifier)
             if processed_identifiers:
-
                 inner_dict["identifiers"] = processed_identifiers
             else:
                 inner_dict.pop("identifiers", None)
 
+        def lookup_person_id(creator):
+            inner_dict = creator.get("person_or_org", {})
+            identifiers = deepcopy(inner_dict.get("identifiers", []))
+            name = None
+
+            # lookup and remove person_id
+            for identifier in reversed(identifiers):
+                if identifier.get("scheme") == "cern":
+                    name = NamesMetadata.query.filter_by(pid=identifier["identifier"]).one_or_none()
+                    # we drop CERN person id identifier
+                    identifiers.remove(identifier)
+
+            creator["person_or_org"]["identifiers"] = identifiers
+            if name:
+                # TODO decide: update name with the latest data from sync
+                # creator["person_or_org"]["family_name"] = name.json["family_name"]
+                # creator["person_or_org"]["given_name"] = name.json["given_name"]
+                ids = deepcopy(creator["person_or_org"]["identifiers"])
+                for identifier in name.json["identifiers"]:
+                    if identifier not in ids and identifier.get("scheme") != "cern":
+                        ids.append(identifier)
+                creator["person_or_org"]["identifiers"] = ids
+                existing_ids = deepcopy(name.json["identifiers"])
+                json_copy = deepcopy(name.json)
+                # update the names vocab to contain other ids
+                for identifier in ids:
+                    if identifier not in existing_ids:
+                        existing_ids.append(identifier)
+                        json_copy["identifiers"] = existing_ids
+                        name.json = json_copy
+                        db.session.add(name)
+                        db.session.commit()
+
         def creators(json, key="creators"):
             _creators = deepcopy(json.get(key, []))
             _creators = list(filter(lambda x: x is not None, _creators))
             for creator in _creators:
                 creator_affiliations(creator)
                 creator_identifiers(creator)
+                lookup_person_id(creator)
```

Comment:
I would move this lookup to the load stage if possible. Also, thinking...isn't that similar to the affiliations stream in the sense of passing by all entities and enrich the names vocabulary? Maybe something to consider in the future to split this in a separate stream that can be run independently...

Reply:
yes, if you remember we discussed this to switch in the future to a separate stream for user related loading, I think it would also go there